schema_utils.py

```diff
@@ -185,6 +185,30 @@ def get_validator(validator_id):
     return _Validators.get(validator_id)
 
 
+def contains_balanced_brackets(expression):
+    """Checks if the given expression contains a balanced bracket sequence.
+
+    Args:
+        expression: str. A math expression (algebraic/numeric).
+
+    Returns:
+        bool. Whether the given expression contains a balanced
+            bracket sequence.
+    """
+    openers, closers = '({[', ')}]'
+    stack = []
+    for character in expression:
+        if character in openers:
+            stack.append(character)
+        elif character in closers:
+            if len(stack) == 0:
+                return False
+            top_element = stack.pop()
+            if openers.index(top_element) != closers.index(character):
+                return False
```

Comment:
Missing an "else raise" at the bottom. Also missing backend tests.

Reply:
I'm sorry, I don't understand why we need an "else raise". If it's not an opening or a closing bracket then we just ignore it. Note that this particular function only checks for balanced parenthesis.

Reply:
Oh, yes, I see. Thanks, no action needed here -- my bad.